packages/contract/src/__test__/call-test-contract/src/main.sw

```diff
@@ -93,4 +104,25 @@ impl TestContract for Contract {
   fn return_context_gas() -> u64 {
     context_gas()
   }
+  fn take_array_string_shuffle(a: [str[3];3]) -> [str[3];3] {
+    [a[2], a[0], a[1]]
+  }
+  fn take_array_string_return_single(a: [str[3];3]) -> [str[3];1] {
+    [a[0]]
+  }
+  fn take_array_string_return_single_element(a: [str[3];3]) -> str[3] {
+    a[0]
+  }
```

Comment:
This is still erroring out;  The return receipt is type `Panic`

Reply:
cc: @mohammadfawaz

Reply:
Hmm interesting.. Any idea whether this is an issue with the compiler or `fuels-ts`? Does `fuels-rs` have the same problem?

Reply:
I wrote some test for `fuels-rs` [here](https://github.com/FuelLabs/fuels-rs/compare/ql/array-tests?expand=1) @digorithm  can you check these look good?

Generating the abi with `cargo run --bin build-test-projects` works, but on `cargo test` I get this error
```
error: proc macro panicked
    --> packages/fuels/tests/harness.rs:2310:5
     |
2310 | /     abigen!(
2311 | |         TestContract,
2312 | |         "packages/fuels/tests/test_projects/array_contract_inputs/out/debug/array_contract_inputs-abi.json"
2313 | |     );
     | |_____^
     |
     = help: message: called `Result::unwrap()` on an `Err` value: InvalidType("Expected parameter type `str[n]`, found `[str[3]; 3]`")
```

I'm on `forc 0.19.0`

Reply:
cc @mohammadfawaz  ^^

Reply:
Created 
https://github.com/FuelLabs/sway/issues/2410
https://github.com/FuelLabs/fuels-rs/issues/505
as followups.  I revisit this and not block the PR